persist/persist.go

```diff
@@ -0,0 +1,60 @@
+package persist
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"os"
+	"sync"
+)
+
+var lock sync.Mutex
+
+// Marshal is a function that marshals the object into an
+// io.Reader.
+// By default, it uses the JSON marshaller.
+var Marshal = func(v interface{}) (io.Reader, error) {
+	b, err := json.MarshalIndent(v, "", "\t")
+	if err != nil {
+		return nil, err
+	}
+	return bytes.NewReader(b), nil
+}
+
+// Save saves a representation of v to the file at path.
+func Save(path string, v interface{}) error {
+	lock.Lock()
```

Comment:
Why do we need a lock here?

Reply:
to avoid having something else mess with it until saving is done

Reply:
For it to have any effect the "something else" must also use the `Mutex`, right?